File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap <Integer,Integer> map=new LinkedHashMap<>();
        int res[]=new int[k];
        for (int i:nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        PriorityQueue<Integer> heap = new PriorityQueue<>((a, b) -> map.get(a) - map.get(b));
        for(int i:map.keySet()){
            heap.add(i);
            if (heap.size()>k){
                heap.poll();
            }
        }
        for (int i=0;i<k;i++){
            res[i]=heap.poll();
            
        }
        return res;

        
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        start=0
        end=k-1
        res=[]
        while end<len(nums) :
            res.append(max(nums[start:end+1]))
            start+=1
            end+=1
        return res

        